include/robot_interfaces/robot_logger.hpp

```diff
@@ -309,45 +303,38 @@ class RobotLogger
                                                                           t1)
                         .count();
 
-// to print the time taken for one block of data to be logged.
-
+                // to print the time taken for one block of data to be logged.
                 std::cout << "Time taken for one block of data to be logged: "
                           << duration << std::endl;
 #endif
             }
         }
-
     }
 
-    /*
-    * call start() to create the thread for the RobotLogger and start logging!
-    */
-
+    /**
+     * @brief Call start() to create the thread for the RobotLogger and start
+     * logging!
+     *
+     * @param The name of the log file.
+     */
     void start(std::string filename)
     {
-
         output_file_name_ = filename;
         thread_->create_realtime_thread(&RobotLogger::write, this);
-
     }
 
-    /*
-    * call stop() when you want to stop logging.
-    */
-
+    /**
+     * @brief Call stop() when you want to stop logging.
+     */
     void stop()
     {
-
-        output_file_.open(output_file_name_, std::ios_base::app);
-
+        stop_was_called_ = true;
         append_robot_data_to_file();
-
-        output_file_.close();
-
+        thread_->join();
```

Comment:
Please put the `thread_->joint()` above the `append_robot_data_to_file()` to prevent potential concurrency conflicts.